IIPS/iip-52.md

```diff
@@ -169,73 +169,70 @@ At this time, the `_from` would be the BTP address of `xcall`.
+but just define operational parts like how to get the proper fee amount before sending the call request, etc.
 
-Here are getter and setter methods for the fixed fees handling. DApps that want to make a call to `sendCallMessage`,
-should query the total fixed fees for the destination network via `totalFixedFees` interface, and then enclose the
-appropriate fees in the method call.
+Here are getter and setter methods for the proper fees handling in `xcall`.
+DApps that want to make a call to `sendCallMessage`, should query the total fee amount for the destination
+network via `getFee` interface, and then enclose the appropriate fees in the method call.
+Note that the protocol fee amount can be get/set via `xcall`, but the relay fee would be obtained from BMC
+which manages BTP network connections.
 
 ```java
 /**
- * Gets the fixed fee for the given network address and type.
- * If there is no mapping to the network address, `default` fee is returned.
+ * Gets the fee for delivering a message to the _net.
+ * If the sender is going to provide rollback data, the _rollback param should set as true.
+ * The returned fee is the sum of the protocol fee and the relay fee.
  *
  * @param _net The network address
- * @param _type The fee type ("relay" or "protocol")
- * @return The fee amount in loop
+ * @param _rollback Indicates whether it provides rollback data
+ * @return the sum of the protocol fee and the relay fee
  */
 @External(readonly=true)
-BigInteger fixedFee(String _net, String _type);
+BigInteger getFee(String _net, boolean _rollback);
 
 /**
- * Gets the total fixed fees for the given network address.
- * If there is no mapping to the network address, `default` fee is returned.
+ * Sets the protocol fee amount.
  *
- * @param _net The network address
- * @return The total fees amount in loop
+ * @param _value The protocol fee amount in loop
+ * @implNote Only the admin wallet can invoke this.
  */
-@External(readonly=true)
-BigInteger totalFixedFees(String _net);
+@External
+void setProtocolFee(BigInteger _value);
 
 /**
- * Sets the fixed fees for the given network address.
- * Only the admin wallet can invoke this.
+ * Gets the current protocol fee amount.
  *
- * @param _net The destination network address
- * @param _relay The relay fee amount in loop
- * @param _protocol The protocol fee amount in loop
+ * @return The protocol fee amount in loop
  */
-@External
-void setFixedFees(String _net, BigInteger _relay, BigInteger _protocol);
+@External(readonly=true)
+BigInteger getProtocolFee();
 
 /**
- * Gets the total accrued fees for the given type.
+ * Sets the address of Fee Handler.
+ * If _addr is null (default), it accrues protocol fees.
+ * If _addr is a valid address, it transfers accrued fees to the address and
+ * will also transfer the receiving fees hereafter.
  *
- * @param _type The fee type ("relay" or "protocol")
- * @return The total accrued fees in loop
+ * @param _addr The address of Fee Handler
```

Comment:
_addr parameter is Optional, how the system will behave if this parameter is absent?

Reply:
That is already described in the javadoc.
> * If _addr is null (default), it accrues protocol fees.
> * If _addr is a valid address, it transfers accrued fees to the address and
> * will also transfer the receiving fees hereafter.